src/server/scripts/Spells/spell_priest.cpp

```diff
@@ -393,6 +418,191 @@ class spell_pri_divine_hymn : public SpellScript
     }
 };
 
+Optional<ObjectGuid> GetDivineImageGUID(Unit* caster)
+{
+    for (Unit* summon : caster->m_Controlled)
+    {
+        if (summon->GetEntry() != NPC_PRIEST_DIVINE_IMAGE)
+            continue;
+
+        return summon->GetGUID();
+    }
+
+    return std::nullopt;
+}
+
+// 392988 - Divine Image
+class spell_pri_divine_image : public AuraScript
+{
+    PrepareAuraScript(spell_pri_divine_image);
+
+    bool Validate(SpellInfo const* /*spellInfo*/) override
+    {
+        return ValidateSpellInfo
+        ({
+            SPELL_PRIEST_DIVINE_IMAGE_SUMMON,
+            SPELL_PRIEST_DIVINE_IMAGE_EMPOWER,
+            SPELL_PRIEST_DIVINE_IMAGE_EMPOWER_STACK
+        });
+    }
+
+    void HandleProc(AuraEffect* aurEff, ProcEventInfo& eventInfo)
+    {
+        Unit* target = eventInfo.GetActor();
+        if (!target)
+            return;
+
+        // Note: if target has an active Divine Image, we should empower it rather than summoning a new one.
+        Optional<ObjectGuid> divineImageGUID = GetDivineImageGUID(target);
+        if (divineImageGUID.has_value())
+        {
+            Unit* divineImage = ObjectAccessor::GetUnit(*target, *divineImageGUID);
+            if (!divineImage)
+                return;
+
+            // Note: Divine Image now teleports near the target when they cast a Holy Word spell if the Divine Image is further than 15 yards away (Patch 10.1.0).
+            if (target->GetDistance(divineImage) > 15.0f)
```

Comment:
Do you want to calculate the 2D Distance or the 3D Distance here? 3D will take Z also into consideration.. 2D will only calculate the straight distance from A to B